src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+
+export const enum LocalStorageKey {
+  tokens = 'tokens',
+  user = 'user',
+}
+
+export type LocalStorageValue = object | null;
+
+export const getLocalStorageValue = (key: LocalStorageKey) => {
```

Comment:
I think typing for this function can be improved
```suggestion
type LocalStorageValueType<T> = 
    T extends LocalStorageKey.tokens ? Tokens :
    T extends LocalStorageKey.tokens ? Users : 
    any;

export const getLocalStorageValue = <T extends LocalStorageKey>(key: T): LocalStorageValueType<T> => {
```

Then we will get automatically typed when getting the local storage value

```ts
const tokens = getLocalStorageValue(LocalStorageKey.tokens) // Token type
const user = getLocalStorageValue(LocalStorageKey.user) // User type
const other_value = getLocalStorageValue("other_value") // any type
```

Reply:
Yeah this is much better 🤩 Updated in 2fa29ca and 72dd143